lark/parsers/lalr_parser.py

```diff
@@ -35,84 +35,123 @@ def parse(self, *args):
         return self.parser.parse(*args)
 
 
-class _Parser:
-    def __init__(self, parse_table, callbacks, debug=False):
+class ParseConf:
+    __slots__ = 'parse_table', 'callbacks', 'start', 'start_state', 'end_state', 'states'
+
+    def __init__(self, parse_table, callbacks, start):
         self.parse_table = parse_table
+
+        self.start_state = self.parse_table.start_states[start]
+        self.end_state = self.parse_table.end_states[start]
+        self.states = self.parse_table.states
+
         self.callbacks = callbacks
-        self.debug = debug
+        self.start = start
 
-    def parse(self, seq, start, set_state=None, value_stack=None, state_stack=None):
-        token = None
-        stream = iter(seq)
-        states = self.parse_table.states
-        start_state = self.parse_table.start_states[start]
-        end_state = self.parse_table.end_states[start]
 
-        state_stack = state_stack or [start_state]
-        value_stack = value_stack or []
+class ParserState:
```

Comment:
I failed, I should have caught this in the original review: classes should be new style and explicit inherit from object.

Reply:
Why is that? Iirc "new style" in Python 2.7 is only for edge features like __getattribute__ and such.

Reply:
And for `type(self)` like used in a `__copy__` method.